rpmlint/checks/SUIDPermissionsCheck.py

```diff
@@ -0,0 +1,224 @@
+                                         f + ' is a file but listed as directory')
+
+                if mode & (stat.S_ISUID | stat.S_ISGID):
+
+                    need_verifyscript = True
+                    msg = '%(file)s is packaged with ' \
+                          'setuid/setgid bits (0%(mode)o)' % \
+                          {'file': f, 'mode': stat.S_IMODE(mode)}
+                    if not stat.S_ISDIR(mode):
+                        self.output.add_info('E', pkg, 'permissions-file-setuid-bit', msg)
+                    else:
+                        self.output.add_info('W', pkg, 'permissions-directory-setuid-bit', msg)
+
+                    if stat.S_ISREG(mode):
+                        if 'shared object' not in pkgfile.magic and 'pie executable' not in pkgfile.magic:
+                            self.output.add_info('E', pkg, 'non-position-independent-executable', f)
+
+                if mode & stat.S_IWOTH:
+                    need_verifyscript = True
+                    self.output.add_info('E', pkg, 'permissions-world-writable',
+                                         f'{f} is packaged with world writable permissions ({oct(mode)})')
+
+            script = not isinstance(pkg, FakePkg) and (pkg[rpm.RPMTAG_POSTIN] or pkg.scriptprog(rpm.RPMTAG_POSTINPROG))
+            found = False
+            if script:
+                for line in script.split('\n'):
+                    if 'chkstat -n' in line and f in line:
+                        found = True
+                        break
+
+                    if 'SuSEconfig --module permissions' in line \
+                            or 'run_permissions is obsolete' in line:
+                        found = True
+                        found_suseconfig = True
+                        break
+
+            if need_verifyscript and \
+                    (f not in self.perms or 'static' not in self.perms[f]):
+
+                if not script or not found:
+                    self.output.add_info('E', pkg, 'permissions-missing-postin',
+                                         f'missing %set_permissions {f} in %post')
+
+                need_set_permissions = True
+                script = pkg[rpm.RPMTAG_VERIFYSCRIPT] or pkg[rpm.RPMTAG_VERIFYSCRIPTPROG]
+
+                found = False
+                if script:
+                    for line in script.split('\n'):
+                        if '/chkstat' in line and f in line:
+                            found = True
+                            break
+
+                if not script or not found:
+                    self.output.add_info('W', pkg, 'permissions-missing-verifyscript',
+                                         f'missing %verify_permissions -e {f}')
+
+        if need_set_permissions:
+            if 'permissions' not in (x[0] for x in pkg.prereq):
+                self.output.add_info('E', pkg, 'permissions-missing-requires',
+                                     "missing 'permissions' in PreReq")
```

Comment:
This sounds wrong, PreReq is obsolete var, now we use Requires(phase): to state the deps.